ex01.exs

```diff
@@ -28,8 +28,25 @@ defmodule Ex01 do
   """
   
   def counter(value \\ 0) do
+    receive do
+      { :next, from } ->
+        send from, { :next_is, value }
```

Comment:
## -2

This is going to seem very picky, but

~~~ elixir
value = value + 1
counter(value)
~~~

expresses (to me, at least) a different meaning to

~~~ elixir
counter(value + 1)
~~~

The first form goes to the trouble of assigning to a variable, which suggests that this value will be needed later. But, later is just the next line, and a simple increment doesn't complicate this line.

So the second form expresses what you are doing more clearly—call myself with an incremented value.

You could also express this using

~~~ elixir
value + 1 |> counter()
~~~